src/ota.rs

```diff
@@ -147,6 +149,7 @@ impl EspOtaUpdate {
 
         self.update_partition = core::ptr::null();
         self.update_handle = 0;
+        self.discarded = true;
```

Comment:
I don't think this field is necessary as `discarded` <=> not in update mode <=> `update_partition` != `null`.